quinn-proto/src/config.rs

```diff
@@ -365,6 +378,72 @@ impl fmt::Debug for TransportConfig {
     }
 }
 
+/// Parameters for controlling the peer's acknowledgement frequency
+///
+/// The parameters provided in this config will be sent to the peer at the beginning of the
+/// connection, so it can take them into account when sending acknowledgements (see each parameter's
+/// description for details on how it influences acknowledgement frequency).
+///
+/// Quinn's implementation follows the fourth draft of the
+/// [QUIC Acknowledgement Frequency extension](https://datatracker.ietf.org/doc/html/draft-ietf-quic-ack-frequency).
+#[derive(Clone, Debug)]
+pub struct AckFrequencyConfig {
+    pub(crate) ack_eliciting_threshold: VarInt,
+    pub(crate) max_ack_delay: Option<Duration>,
+    pub(crate) reordering_threshold: VarInt,
+}
+
+impl AckFrequencyConfig {
+    /// The number of ack-eliciting packets the remote peer may receive without immediately sending
+    /// an ACK.
+    ///
+    /// The remote peer should send at least one ACK frame when more than this number of
+    /// ack-eliciting packets have been received. A value of 0 results in a receiver immediately
+    /// acknowledging every ack-eliciting packet.
+    ///
+    /// Defaults to 1, which sends ACK frames for every other ack-eliciting packet.
+    pub fn ack_eliciting_threshold(&mut self, value: VarInt) -> &mut Self {
+        self.ack_eliciting_threshold = value;
+        self
+    }
+
+    /// The maximum amount of time that the remote peer waits before sending an ACK when the
+    /// ack-eliciting threshold hasn't been reached.
+    ///
+    /// The provided `max_ack_delay` will be clamped to be at least the peer's `min_ack_delay`
+    /// transport parameter.
+    ///
+    /// Defaults to `None`, in which case the peer's original `max_ack_delay` will be used.
+    pub fn max_ack_delay(&mut self, value: Option<Duration>) -> &mut Self {
+        self.max_ack_delay = value;
+        self
+    }
+
+    /// The amount of out-of-order packets that will trigger a peer to send an ACK, without waiting
+    /// for `ack_eliciting_threshold` to be exceeded or for `max_ack_delay` to be elapsed.
```

Comment:
> Clearly we can't configure the peer to do anything.

The way ACK frequency works in the current draft is that you actually "configure" the peer, telling it which parameters it should use to regulate sending of ACKs. Specifically, we tell the peer which `max_ack_delay`, `ack_eliciting_threshold`, and `reordering_threshold` to use.

The doc comment to the `AckFrequencyConfig` offers valuable context on how this works:

```rust
/// Parameters for controlling the peer's acknowledgement frequency
///
/// The parameters provided in this config will be sent to the peer at the beginni
/// connection, so it can take them into account when sending acknowledgements (se
/// description for details on how it influences acknowledgement frequency).
```

But maybe you are already aware of all this, and it's just the wording that feels odd? IMO with the above comment and the fact that a compliant peer is supposed to obey the `ACK_FREQUENCY` frame, I think it's OK to imply that we are configuring the peer.

Reply:
Right -- it's mostly the wording that I'm objecting to here. I'd like us to be cognizant of the fact that we can request the peer to do things and we can expect the peer to do things a certain way, but "will trigger" seems too strong to me -- to some extent the peer is unreliably and we have to do the right thing even if it doesn't.

> Regarding style, is it just the first line that shouldn't end with a "."?

The idea is that the first line is usually a sentence fragment rather than a sentence (a sentence without a subject, a verb, or it doesn't include a "complete thought"). Because the rest of the docstring is not subject to the same conciseness constraints, they can usually be more fully formed.

Reply:
Thanks, I have reworded the comments and made sure the first lines are shorter